clipper_admin/clipper_admin/clipper_admin.py

```diff
@@ -1187,3 +1190,68 @@ def stop_all(self):
         """
         self.cm.stop_all()
         logger.info("Stopped all Clipper cluster and all model containers")
+
+    def test_predict_function(self, query, func, input_type):
+        """Tests that the user's function has the correct signature and can be properly saved and loaded.
+
+        The function should take a dict request object like the query frontend expects JSON, 
+        the predict function, and the input type for the model.
+
+        Parameters
+        ----------
+        query: JSON or list of dicts
+            Inputs to test the prediction function on.
+        func: function
+            Predict function to test.
+        input_type: str
+            The input_type to be associated with the registered app and deployed model.
+            One of "integers", "floats", "doubles", "bytes", or "strings".
+        """
+        query_data = list(x for x in list(query.values()))
```

Comment:
What's going on here?

Reply:
You should be checking the JSON key as well, to ensure that their input is properly formatted

Reply:
Assuming that the query is in a JSON/dict structure, I'm getting the values of each key. I'm then checking to make sure the values are of the right input_type before then converting it into the respective numpy array of right dtype. What formatting of the key input needs to be checked?

Reply:
If the user wants to provide a single input, the key should be `"input"`. If they want to provide a list of inputs, the key should be `"input_batch"`. See [the website](http://clipper.ai/tutorials/basic_concepts/) for details.

Reply:
Got it. Thanks for the clarification.